src/env/emulated_syscalls/linux_syscalls.cpp

```diff
@@ -776,6 +778,102 @@ FunctionCallback::return_t sys_linux_exit(
     return std::monostate();
 }
 
+// ssize_t getrandom(void *buf, size_t buflen, unsigned int flags);
+FunctionCallback::return_t sys_linux_getrandom(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    addr_t buf = args[0].as_uint(*engine.vars);
+    size_t buflen = args[1].as_uint(*engine.vars);
+    cst_t res;
+
+    // TODO: generate proper random bytes, if needed
+    std::string rand(buflen, 'A'); 
+    engine.mem->write_buffer(buf, (uint8_t*)rand.c_str(), buflen);
+    res = buflen;
+    return res;
+}
+
+// pid_t getpid(void);
+FunctionCallback::return_t sys_linux_getpid(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    return engine.process->pid;
+}
+
+// uid_t getuid(void);
+FunctionCallback::return_t sys_linux_getuid(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    return engine.process->ruid;
+}
+
+// uid_t geteuid(void);
+FunctionCallback::return_t sys_linux_geteuid(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    return engine.process->euid;
+}
+
+// pid_t gettid(void);
+FunctionCallback::return_t sys_linux_gettid(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    // TODO: implement
+    // In a single-threaded process, the thread ID is equal to the process ID
+    return sys_linux_getpid(engine, args);
+}
+
+// long set_tid_address(int *tidptr);
+FunctionCallback::return_t sys_linux_set_tid_address(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    // TODO: implement
+    // Always return the caller's thread ID
+    return sys_linux_gettid(engine, args);
+}
+
+// long set_robust_list(struct robust_list_head *head, size_t len);
+FunctionCallback::return_t sys_linux_set_robust_list(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    // TODO: implement
+    return -ERR_ENOSYS;
+}
```

Comment:
Need to look a bit more into how robust futexes work, but since we don't support multi-threading out-of-the-box it should be OK to have a dummy implementation, or even just return `0` to fake success.